dto/src/main/java/ca/on/oicr/gsi/runscanner/dto/dragen/BCLConvert.java

```diff
@@ -0,0 +1,197 @@
+package ca.on.oicr.gsi.runscanner.dto.dragen;
+
+import ca.on.oicr.gsi.runscanner.dto.dragen.samplesheet.Samplesheet;
+import ca.on.oicr.gsi.runscanner.dto.dragen.samplesheet.SamplesheetBCLConvertSection;
+import ca.on.oicr.gsi.runscanner.dto.dragen.samplesheet.SamplesheetBCLConvertSection.SamplesheetBCLConvertDataEntry;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BCLConvert {
+  private static final Logger log = LoggerFactory.getLogger(BCLConvert.class);
+  private final Samplesheet samplesheet;
+  private final File rootDir;
+  private DragenWorkflowAnalysis result;
+  private boolean isOk = false;
+
+  public BCLConvert(Samplesheet sheet, File dir) throws IOException {
+    this.samplesheet = sheet;
+    this.rootDir = dir;
+    process();
+  }
+
+  public boolean isOk() {
+    return isOk;
+  }
+
+  public DragenWorkflowAnalysis getResult() throws IOException {
+    if (result == null) process();
+    return result;
+  }
+
+  private void process() throws IOException {
```

Comment:
Dtos module should not contain any processing

Reply:
Moved BCLConvert back to the processor package and made it return a dto instead